src/render.rs

```diff
@@ -41,18 +51,89 @@ pub fn emit_commandline(
         format!("oci://{image}")
     };
 
-    let mut cli = [
-        "kubecfg",
-        "show",
-        "--alpha",
-        "--reorder=server",
-        &entrypoint,
-        "--overlay-code-file",
-        &format!("appInstance_={overlay_file}"),
-    ]
-    .iter()
-    .map(|s| s.to_string())
-    .collect::<Vec<String>>();
+    let mut cli: Vec<String> = vec![];
+    let overlay_path = std::fs::canonicalize(overlay_file).unwrap();
+    let overlay_file_name = std::path::PathBuf::from(overlay_path.file_name().unwrap());
+    let user_home = home_dir().expect("unable to retrieve home directory");
+    let docker_config =
+        env::var("DOCKER_CONFIG").unwrap_or(format!("{}/.docker", user_home.display()));
+    let kube_config =
+        env::var("KUBECONFIG").unwrap_or(format!("{}/.kube/config", user_home.display()));
+
+    if is_local {
+        let temp_file = NamedTempFile::new().expect("unable to create temporary file");
+        serde_yaml::to_writer(&temp_file, app_instance).unwrap();
+
+        let meta = metadata::fetch_package_config(&temp_file.path().to_string_lossy())
```

Comment:
This _seems_ like the right thing to do, but maybe there's a better way.

The need for a temporary file is to leverage the pre-existing `fetch_package_config` from the `metadata` crate, rather than having our own implementation of it, I don't think that is particularly necessary here since we simply want to extract the kubecfg image/version out of it.

Reply:
the `fetch_package_config` method does two things:

1. reads and decodes a file into an `AppInstance` structure
2. calls oci:fetch_package_config with that `AppInstance` structure and some local docker auth

Here we already have an `AppInstance` struct and we just want to leverage the fact that the `metadata::fetch_package_config` function deals with the details around handling local docker auth.

I suggest we make a little refactoring of the `metadata::fetch_package_config` function and split it into two parts, so we can skip the part where the file is read from the filesystem.

https://github.com/kubecfg/kubit/pull/179

once that's merged you can rebase this PR on top of that and make use of that function